exec/cnex/cnex.c

```diff
@@ -154,45 +196,113 @@ int main(int argc, char* argv[])
                         exec->diagnostics.total_opcodes,
                         exec->stack->max + 1,
                         exec->stack->top,
-                        exec->diagnostics.callstack_max_height + 1, 
-                        exec->callstacktop, 
-                        exec->object->global_size, 
+                        exec->diagnostics.callstack_max_height + 1,
+                        exec->callstacktop,
+                        exec->module[0].bytecode->global_size,
                         exec->framestack->max,
                         (((float)exec->diagnostics.time_end - exec->diagnostics.time_start) / CLOCKS_PER_SEC) * 1000
         );
     }
 
-    global_shutdown();
     exec_freeExecutor(exec);
-    bytecode_freeBytecode(pModule);
+    global_shutdown();
 
-    free(bytecode);
+    free(gOptions.pszExecutablePath);
 
     return 0;
 }
 
 int exec_run(TExecutor *self, BOOL enable_assert)
 {
-    self->ip = (uint32_t)self->object->codelen;
-    invoke(self, 0);
     self->enable_assert = enable_assert;
+    self->ip = (uint32_t)self->module->bytecode->codelen;
+    invoke(self, self->modules[0], 0);
+
+    for (unsigned int m = 1; m < self->module_count; m++) {
+        invoke(self, self->modules[m], 0);
+    }
 
     exec_loop(self);
+
     if (gOptions.ExecutorDebugStats) {
         assert(isEmpty(self->stack));
     }
-
     return 0;
 }
 
-TExecutor *exec_newExecutor(TBytecode *object)
+// This function locates, and reads the binary byte code in from a compiled Neon module.
+uint8_t *exec_readBytecode(const char *name, unsigned int *code_size)
+{
+    // ToDo: Implement %neonpath% environment variable, as well as ".neonpath" path file.
+    int pass = 0;
+    char filename[256];
+    // For this first case, look in the cwd for the module.
+    sprintf(filename, "%s.neonx", name);
+    FILE *fp = NULL;
+
+load:
+    fp = fopen(filename, "rb");
+    if (fp == NULL) {
+        switch (pass) {
```

Comment:
This "pass" stuff and the `goto` is confusing to read. Not acceptable. Find a better way.

Reply:
Removed.  Fixed with support.c, which encapsulates all of the locating and loading modules.

Reply:
I'm not seeing the fix here.